packages/core/src/components/icon/icon.md

```diff
@@ -4,15 +4,62 @@
     See [**Icons**](#icons) for a searchable list of all available UI icons.
 </div>
 
+<div class="pt-callout pt-intent-primary pt-icon-info-sign">
+    <h5>SVG icons in 2.0</h5>
+    Blueprint 2.0 introduced SVG icon support throughout the ecosystem, and moved icon resources to a separate
+    __@blueprintjs/icons__ package. The `Icon` component now renders SVG paths and the icon classes are no longer
+    used by any Blueprint React component. Icon font support has been preserved but should be considered legacy,
+    and we plan to remove the icon fonts in a future major version.
+</div>
+
 This section describes two ways of using the UI icon font: via CSS or via React component.
 
 Many Blueprint components provide an `iconName` prop, which supports both the
 full name `pt-icon-projects` and the short name `projects`.
 
 @reactExample IconExample
 
+@## JavaScript API
+
+The `Icon` component is available in the __@blueprintjs/core__ package.
+Make sure to review the [general usage docs for JS components](#blueprint.usage).
+
+Use the `<Icon>` component to easily render __SVG icons__ in React. The `iconName` prop is typed
+such that editors can offer autocomplete for known icon names. The optional `iconSize` prop ensures
+you'll never forget a sizing class and clarifies the expected width and height of the icon element.
+The component also accepts all valid HTML props for an `<svg>` element.
+
+Data files in the __@blueprintjs/icons__ package provide SVG path information for Blueprint's 300+ icons
+for 16px and 20px grids. The `iconName` prop dictates which SVG is rendered, and `iconSize` determines
```

Comment:
another unnecessary comma here

Reply:
i actually thought long and hard about this one